events/EventQueue.h

```diff
@@ -139,6 +148,35 @@ class EventQueue : private mbed::NonCopyable<EventQueue> {
      */
     bool cancel(int id);
 
+    /** Cancel an in-flight user allocated event
+     *
+     *  Attempts to cancel an UserAllocatedEvent referenced by its address
+     *  It is safe to call cancel after an event has already been dispatched.
+     *
+     *  Event must be valid i.e. event must have not finished executing
+     *  and must have been bound to this queue.
+     *
+     *  The cancel function is IRQ safe.
+     *
+     *  If called while the event queue's dispatch loop is active in another thread,
+     *  the cancel function does not guarantee that the event will not execute after it
+     *  returns, as the event may have already begun executing. A call made from
+     *  the same thread as the dispatch loop will always succeed with a valid id.
+     *
+     *  @param event    Address of the event
+     *  @return         true  if event was successfully cancelled
+     *                  false if event was not cancelled (invalid queue or executing already begun)
+     */
+    template<typename... Args>
+    bool cancel(UserAllocatedEvent<Args...> *event)
+    {
+        MBED_ASSERT(event->_equeue == &_equeue);
+        if (event->_equeue != &_equeue) {
```

Comment:
It doesn't really make sense to assert something and test for it. Stick to one or the other.

Reply:
fixed